### PR TITLE
Use black, isort, ruff, pre-commit

### PR DESCRIPTION
**Pull request by jzuhone**
## Description

This PR only implements [black](https://github.com/psf/black), [isort](https://pycqa.github.io/isort/), and [ruff](https://github.com/charliermarsh/ruff) formatting with [pre-commit](https://pre-commit.com), using the template from https://github.com/sot/skare3/pull/983. There are no code changes that should affect results in any way. 

Unfortunately, it ends up touching nearly every file. ~~I still need to run tests.~~ All tests pass.

@taldcroft, any pointers for how you've had similar PRs reviewed? 

## Interface impacts

None.

### Unit tests

- [ ] No unit tests
- [x] Mac
- [ ] Linux
- [ ] Windows

### Functional tests

Purposefully triggered error and warning conditions and verified that the changes did not affect how these are reported.

### CODE REVIEW

**Comment by taldcroft** [comment]
@jzuhone - generally speaking we have not done anything more than cursory inspection of the black/isort changes since they are non-impacting as far as the actual code. Essentially if unit tests pass then it is good from the code perspective.

I note that the new black and flake8 checks are not reflected in the status at the bottom, i.e. it doesn't look like these GitHub actions are actually being run. You might need to look in the repo settings. There is something in the Branch protection rules to require actions to be successful for merge. Honestly sometimes these checks confuse me, and for instance in the kadi repo now there are black and flake8 checks but it looks like it is only running flake8.

**Comment by taldcroft** [comment]
@javierggt - you understand GitHub actions better than me, do you have any insight about why they aren't being run yet in this repo?

**Comment by javierggt** [comment]
Github actions run only from master. Not only that, they run from master where the branch starts, which is kind of a pain.

**Comment by jzuhone** [comment]
@javierggt Ah, so we have to get this merged first, got it.

File: setup.cfg [hunk 1]
```diff
@@ -0,0 +1,9 @@
+[flake8]
+max-line-length = 100
+exclude = doc,
```
Comment (taldcroft) [inline setup.cfg:3]: Coincidentally I just discovered yesterday the flake8 [per-file-ignores](https://flake8.pycqa.org/en/latest/user/options.html#cmdoption-flake8-per-file-ignores) option that let's you exclude just F401 in `__init__.py` files.

**Review by taldcroft — APPROVED** [review]
The configuration updates here look about right to me. I made one minor suggestion.

**Comment by taldcroft** [comment]
Not directly related to this, but following the astropy lead I'm playing around with using ruff instead of flake8. I think this is the future of Python linting, ruff is really amazing. One driver for me was that I got annoyed that flake8 has refused to support configuration in `pyproject.toml`, as you noticed here with needing to add the new `setup.cfg` file.

**Comment by taldcroft** [comment]
See https://github.com/sot/cxotime/pull/33/commits/e0d530baf6d01f4b32734d84e1cbac43389a809a

**Review by Gregg140 — APPROVED** [review]
Looks ok. I mostly checked the testing and ATC bits that I'm involved with (e.g. dpa_check).

**Comment by taldcroft** [comment]
@jzuhone - I see you got the ruff bug. I got a :smile: this morning at the commit messages.

**Comment by jzuhone** [comment]
@taldcroft @Gregg140 I implemented ruff, which required more changes, some of which were not completely trivial, but I've done some extra functional testing and re-ran the regression testing. 

Relevant commits are https://github.com/acisops/acis_thermal_check/pull/59/commits/c9a64d9d0964de9f9dfcffedfc157ac5e850a236 through https://github.com/acisops/acis_thermal_check/pull/59/commits/720cf35494bb55c1bc793e19a5dabfd8ab2f7aa0

**Comment by jzuhone** [comment]
@taldcroft let me know if you think this is good.

**Comment by jzuhone** [comment]
@taldcroft @Gregg140 circling back to this to see if there are any final issues. 

Relevant commits are https://github.com/acisops/acis_thermal_check/commit/c9a64d9d0964de9f9dfcffedfc157ac5e850a236 through https://github.com/acisops/acis_thermal_check/commit/720cf35494bb55c1bc793e19a5dabfd8ab2f7aa0

**Comment by Gregg140** [comment]
Looks ok to me.

Gregg

**Pull request merged**